libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -0,0 +1,757 @@
+    };
+    auto parsed_options = std::vector<std::string>{};
+    auto header = std::string{};
+    for (const auto& prefix : prefix_options) {
+      ++it;
+      if (not *it or it == lines.end()) {
+        return caf::make_error(ec::syntax_error,
+                               fmt::format("Zeek TSV file header ended too "
+                                           "early - aborting"));
+      }
+      header = (*it).value();
+      auto pos = header.find(prefix);
+      if (pos != 0)
+        return caf::make_error(ec::syntax_error,
+                               fmt::format("Invalid header line: prefix '{}' "
+                                           "not beginning at line "
+                                           "beginning or not found",
+                                           prefix));
+      pos = header.find(sep);
+      if (pos == std::string::npos)
+        return caf::make_error(ec::syntax_error,
+                               fmt::format("Invalid header line: separator "
+                                           "'{}' not found",
+                                           sep_option));
+      if (pos + 1 >= header.size())
+        return caf::make_error(ec::syntax_error,
+                               fmt::format("Missing Zeek TSV header line "
+                                           "content: {}",
+                                           header));
+      parsed_options.emplace_back(header.substr(pos + 1));
+    }
+    set_sep = std::string{parsed_options[0][0]};
+    empty_field = parsed_options[1];
+    unset_field = parsed_options[2];
+    path = parsed_options[3];
+    fields_str = parsed_options[5];
+    types_str = parsed_options[6];
+    fields = detail::split(fields_str, sep);
+    types = detail::split(types_str, sep);
+    if (fields.size() != types.size())
+      return caf::make_error(ec::syntax_error,
+                             fmt::format("Zeek TSV header types mismatch: "
+                                         "expected {} fields but got {}",
+                                         fields.size(), types.size()));
+
+    std::vector<struct record_type::field> record_fields;
+    for (auto i = size_t{0}; i < fields.size(); ++i) {
+      auto t = parse_type(types[i]);
+      if (!t)
+        return std::move(t.error());
+      record_fields.push_back({
+        std::string{fields[i]},
+        *t,
+      });
+    }
+    name = std::string{type_name_prefix} + path;
+    // If a congruent type exists in the module, we give the type in the
+    // module precedence.
+    auto record_schema = record_type{record_fields};
+    record_schema = detail::zeekify(record_schema);
+    output_slice_schema = {};
+    for (const auto& ctrl_schema : ctrl.schemas()) {
+      if (ctrl_schema.name() == name) {
+        auto is_castable = can_cast(record_schema, ctrl_schema);
+        if (!is_castable) {
+          VAST_WARN("zeek-reader ignores incompatible schema '{}' from schema "
```

Comment:
```suggestion
          VAST_WARN("zeek-tsv parser ignores incompatible schema '{}' from schema "
```

Reply:
Fixed in https://github.com/tenzir/vast/pull/3114/commits/d01e63a216f6cf0cb80a82adb3a4ae4e89a9286c